app/src/main/java/org/simple/clinic/summary/addphone/AddPhoneNumberDialog.kt

```diff
@@ -73,9 +77,22 @@ class AddPhoneNumberDialog : AppCompatDialogFragment(), AddPhoneNumberUi {
         .share()
   }
 
+  private val delegate: MobiusDelegate<AddPhoneNumberModel, AddPhoneNumberEvent, AddPhoneNumberEffect> by unsafeLazy {
+    val uiRenderer = AddPhoneNumberUiRender(this)
+
+    MobiusDelegate.forActivity(
+        events = dialogEvents.ofType(),
+        defaultModel = AddPhoneNumberModel.create(),
+        update = AddPhoneNumberUpdate(),
+        effectHandler = effectHandlerFactory.create(this).build(),
+        modelUpdateListener = uiRenderer::render
+    )
+  }
+
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
     TheActivity.component.inject(this)
+    delegate.onRestoreInstanceState(savedInstanceState)
```

Comment:
Are we doing this here because we shouldn't expect dialog to work like activities and did not use onRestoreInstanceState?

Reply:
Yeah, the way we restore is different in fragments and activities compared to views.